### PR TITLE
[AC-8974]: Add user note models

### PR DESCRIPTION
**Pull request by kenware**
[AC-8974](https://masschallenge.atlassian.net/browse/AC-8974)

How to test:
- Test is passing
- Definition of done matches [AC-8974](https://masschallenge.atlassian.net/browse/AC-8974)
- Check sibling PR https://github.com/masschallenge/accelerate/pull/3016

### CODE REVIEW

**Comment by jonkiparsky** [comment]
One high-level issue here: the name of this model is "Note" but I suspect we're going to want to be more specific. In the future we're likely to want to offer notes about Startups, Partners, and other sorts of objects that may not even exist in the system now. 

I would ask that this class be called "UserNote" to allow room in the "Note" namespace for other specific note types, or to migrate this to a more generic "Note" class.

File: accelerator/tests/factories/note_factory.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+from factory import Sequence, SubFactory
+
+from factory.django import DjangoModelFactory
+
+from simpleuser.tests.factories.user_factory import UserFactory
+
+Note = swapper.load_model('accelerator', 'UserNote')
+
+
+class NoteFactory(DjangoModelFactory):
```
Comment (jonkiparsky) [inline accelerator/tests/factories/note_factory.py:12]: rename as UserNoteFactory, consistent with model name

File: accelerator/models/user_note.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import swapper
+from accelerator_abstract.models import BaseUserNote
+
+
+class UserNote(BaseUserNote):
+    class Meta(BaseUserNote.Meta):
+        swappable = swapper.swappable_setting(
+            BaseUserNote.Meta.app_label, "Note"
```
Comment (iankigen) [inline accelerator/models/user_note.py:8]: `"Note"` should be updated to `UserNote`

Comment (kenware) [reply accelerator/models/user_note.py:8]: Thought I pushed

File: accelerator_abstract/models/base_user_note.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+from django.db import models
+
+from accelerator_abstract.models.accelerator_model import AcceleratorModel
+
+
+class BaseUserNote(AcceleratorModel):
+    user = models.ForeignKey(settings.AUTH_USER_MODEL,
+                             related_name="user_notes",
+                             on_delete=models.CASCADE)
+    manager = models.ForeignKey(settings.AUTH_USER_MODEL,
```
Comment (iankigen) [inline accelerator_abstract/models/base_user_note.py:12]: this should be renamed to `author`

**Review by iankigen — APPROVED** [review]
Code/QA looks good

File: accelerator/tests/factories/user_note_factory.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import swapper
+
+from factory import Sequence, SubFactory
```
Comment (jonkiparsky) [inline accelerator/tests/factories/user_note_factory.py:3]: multiple imports should be on separate lines
```
from factory import (
    Sequence, 
    SubFactory,
)
```

File: accelerator/tests/factories/user_note_factory.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import swapper
```
Comment (jonkiparsky) [inline accelerator/tests/factories/user_note_factory.py:1]: @iankigen is it possible that we can use what we learned from CAM Phase 1 and avoid Swappering this new model?

File: accelerator/tests/factories/user_note_factory.py [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+from factory import Sequence, SubFactory
+
```
Comment (jonkiparsky) [inline accelerator/tests/factories/user_note_factory.py:4]: I would typically not have line breaks between all imports. It's nice to have a separation between "their" imports and "ours", but not between each line of imports.

**Review by jonkiparsky — APPROVED** [review]
This is basically fine - a few relatively trivial style notes which I would like to see addressed, but it seems like we're basically there. 

(but note change suggested on the sibling PR)

File: accelerator/tests/factories/user_note_factory.py (hunk 5 above)
Comment (iankigen) [reply accelerator/tests/factories/user_note_factory.py:1]: yeah, it should be possible to avoid using swapper.

**Review by iankigen — COMMENTED** [review]
Looks good

**Comment by jonkiparsky** [comment]
verified unswapperized models work in django shell.

**Pull request merged**